lib/chef/provider/package/yum.rb

```diff
@@ -1081,22 +1082,24 @@ def load_current_resource
                 @new_resource.version($2)
               end
             end
-          end
-
-          if @new_resource.version
-            new_resource = "#{@new_resource.package_name}-#{@new_resource.version}#{yum_arch}"
+            @candidate_version << @new_resource.version
+            installed_version << @yum.installed_version(@current_resource.package_name, arch)
```

Comment:
So at the moment we don't support multi-package *and* `source`, and I'm mostly OK with that, but we should probably check for that in `requirements`, rather than blowing up here. Assuming that, this should work.